source/fundamentals/connection/network-compression.txt

```diff
@@ -0,0 +1,91 @@
+
+.. note::
+
+   Applications that require Snappy or Zstandard compression must
+   :ref:`add explicit dependencies <compression-dependencies>` for those
+   algorithms.
+
+.. _enable-compression:
+
+Specify Compression Algorithms
+------------------------------
+
+You can enable compression for the connection to your MongoDB instance
+by specifying the algorithms in one of two ways: adding the parameter to your
+connection string using ``ConnectionString`` or by calling the method in the
+``MongoClientSettings.Builder`` class.
+
+.. tabs::
+
+   .. tab:: ConnectionString
+      :tabid: connectionstring
+
+      To enable compression using the `ConnectionString <{+api+}/apidocs/mongodb-driver-core/com/mongodb/ConnectionString.html>`__,
+      add the parameter ``compressors`` in the connection string passed to
+      ``MongoClients.create()``. You can specify one or more compression
+      algorithms, separating them with commas:
+
+      .. code-block:: java
+
+         ConnectionString connectionString = new ConnectionString("mongodb+srv://<user>:<password>@<cluster-url>/?compressors=snappy,zlib,zstd");
+         MongoClient mongoClient = MongoClients.create(connectionString);
+
+      Specify compression algorithms using the following strings:
+
+      - "snappy" for `Snappy <https://google.github.io/snappy/>`__ compression
+      - "zlib" for `Zlib <https://zlib.net/>`__ compression
+      - "zstd" for `Zstandard <https://github.com/facebook/zstd/>`__ compression
+
+   .. tab:: MongoClientSettings
+      :tabid: mongoclientsettings
+
+      To enable compression using the `MongoClientSettings <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.html>`__,
+      pass the `compressorList() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#compressorList(java.util.List)>`__
+      builder method a list of `MongoCompressor <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoCompressor.html>`__
+      instances. You can specify one or more compression algorithms in the list:
+
+      .. code-block:: java
+         :emphasize-lines: 2-4
+
+         MongoClientSettings settings = MongoClientSettings.builder()
+              .compressorList(Arrays.asList(MongoCompressor.createSnappyCompressor(),
+                                            MongoCompressor.createZlibCompressor(),
+                                            MongoCompressor.createZstdCompressor()))
+              .build();
+         MongoClient client = MongoClients.create(settings);
+
+.. _compression-dependencies:
+
+Compression Algorithm Dependencies
+----------------------------------
+
+The JDK supports `Zlib <https://zlib.net/>`__ compression natively, but
+`Snappy <https://google.github.io/snappy/>`__ and
+`Zstandard <https://github.com/facebook/zstd/>`__ depend on open source
+implementations. See
+`snappy-java <https://github.com/xerial/snappy-java>`__ and
+`zstd-java <https://github.com/luben/zstd-jni>`__ for details.
```

Comment:
issue: Not your fault but I just closed https://github.com/mongodb/docs-java/pull/187 that explicitly states the versions of these the driver tests against.

Reply:
Got it. I'll handle in the merge.